File: 프로그래머스/0/120853. 컨트롤 제트/컨트롤 제트.js
function solution(s) {
  const stack = [];
  const tokens = s.split(' ');

  for (const token of tokens) {
    if (token === 'Z') {
      stack.pop();          // 직전에 더했던 숫자 제거
    } else {
      stack.push(Number(token));
    }
  }

  return stack.reduce((acc, v) => acc + v, 0);
}

File: 프로그래머스/0/120853. 컨트롤 제트/컨트롤 제트.java
import java.util.Stack;

class Solution {
    public int solution(String s) {
        Stack<Integer> stack = new Stack<>();

        String[] tokens = s.split(" ");

        for (String token : tokens) {
            if (token.equals("Z")) {
                stack.pop(); // 직전에 더했던 숫자 제거
            } else {
                stack.push(Integer.parseInt(token));
            }
        }

        int answer = 0;
        while (!stack.isEmpty()) {
            answer += stack.pop();
        }

        return answer;
    }
}